src/_CPlugin_Helper.h

```diff
@@ -629,7 +629,31 @@ bool client_available(WiFiClient& client) {
 bool send_via_http(const String& logIdentifier, WiFiClient& client, const String& postStr, bool must_check_reply) {
   bool success = !must_check_reply;
   // This will send the request to the server
-  client.print(postStr);
+  byte written = client.print(postStr);
+  if (written != (postStr.length()%256)) {
```

Comment:
Why compare to length modulo 256?
If the `postStr` is longer than 255, this will always fail.

Reply:
It's the other way round. The return value is only one byte (0-255). Some tests showed  that the return value is always the size of the last byte sent if the size to send is more than one byte! That's why the modulo 256 is needed, otherwise it will always fall! You can test it if you enable the debug log! It shows total bytes to send as well as the return value!

Reply:
Ah, I get it, the returned value is an unsigned 8 bit value?

Reply:
Exactly!

Reply:
maybe it is bug in core?

Reply:
@uzi18 see #1957 and #1987 for details and solution about the issue.

Reply:
@TD-er the patch runs for over 24h on my 30 nodes without issues, however also the error condition is never met (anymore) since applying the distance fix... so layer 2 seems stable...

Reply:
I mean Arduino core client interface bug

Reply:
sorry, I don't understand which bug you're referring to.

if you have studied the two links above, this PR is not about fixing a bug but adding some debug information and a safety check around the succesfull completion of a library call that have just been ignored up to now.

Reply:
I think wifi client should return 16 bit integer not uint8_t

Reply:
https://github.com/esp8266/Arduino/blob/8ae0746e4aeaf7c2a8881831f370b40347e47a50/libraries/ESP8266WiFi/src/WiFiClient.h#L67

Reply:
the [reference page](https://www.arduino.cc/en/Reference/WiFiClientPrint) says different... but feel free to suggest that to the esp core guys... until then, this implementation is absolutely correct.... and I know you like reference implementations 😄

Reply:
You don't understand - available() returns int, read() returns int, print and write returns size_t.

Reply:
May be. but completely irreleant for this PR..

Up to @TD-er to include it or not...

Reply:
It is a bit hard to find in the source, what is actually being returned.
I think it is this part of the code:
https://github.com/esp8266/Arduino/blob/561426c0c77e9d05708f2c4bf2a956d3552a3706/libraries/ESP8266WiFi/src/include/ClientContext.h#L437-L467

The `_written` is a `size_t`, but that doesn't matter of course when it is later handled by a function only returning a byte.
Did you send post strings > 255 bytes?

Reply:
more like here: https://github.com/esp8266/Arduino/blob/561426c0c77e9d05708f2c4bf2a956d3552a3706/libraries/ESP8266WiFi/src/include/ClientContext.h#L364

Reply:
@uzi18 That will call `_write_from_source`, which is what I pointed to.

I was thinking, since we all agree there is some discrepancy between (apparently) the current implementation and what is logical/desired.
So it may be making sense if it will be changed later on.
We can have some kind of generic compare for this which checks for the `size % 256` and if the written value is > 256, just compare directly.

Or at least place a reference in comments to this discussion. If it will get changed in the future, you can take a lot of time looking for errors like these.